### PR TITLE
Feature/chandan/appdev 77

### PR DESCRIPTION
**Pull request by chandanvarma009**
- budget goal picker fixed
- timeZone and timeBudget update call implemented
- TimeZone picker fixed, Validations, from-to issues fixed
- TimePicker issues fixed

### CODE REVIEW

File: Yona/Yona/Challenges/TimeBucketChallenges.swift [hunk 1]
```diff
@@ -203,6 +204,13 @@ class TimeBucketChallenges: UIViewController,UIScrollViewDelegate {
                     Loader.Hide(self)
                 }
                 self.activityCategoriesArray = activities!
+                if self.activityCategoriesArray.count > 0 {
+                    self.addNewGoalButton.hidden = false
+                } else {
+                    dispatch_async(dispatch_get_main_queue()) {
+                    self.addNewGoalButton.hidden = true
+                    }
+                }
```
Comment (vasakinnar) [inline Yona/Yona/Challenges/TimeBucketChallenges.swift]: @chandanvarma009  no need for if else. can't you just give self.activityCategoriesArray.count > 0 to self.addNewGoalButton.hidden

File: Yona/Yona/Challenges/TimeBucketChallenges.swift [hunk 2]
```diff
@@ -260,6 +268,13 @@ class TimeBucketChallenges: UIViewController,UIScrollViewDelegate {
     
     // MARK: - Tapgesture category view tap events
     func categoryTapEvent(sender: UITapGestureRecognizer? = nil) {
+        if self.activityCategoriesArray.count > 0 {
+            self.addNewGoalButton.hidden = false
+        } else {
+            dispatch_async(dispatch_get_main_queue()) {
+                self.addNewGoalButton.hidden = true
+            }
+        }
```
Comment (vasakinnar) [inline Yona/Yona/Challenges/TimeBucketChallenges.swift]: @chandanvarma009 no need for if else. can't you just give self.activityCategoriesArray.count > 0 to self.addNewGoalButton.hidden

File: Yona/Yona/Challenges/TimeFrameTimeZoneChallengeViewController.swift [hunk 3]
```diff
@@ -29,16 +34,59 @@ class TimeFrameTimeZoneChallengeViewController: UIViewController {
...
+            if self.picker?.cancelButtonTitle.title == "Prev" {
+                self.isFromButton = true
+                self.picker?.pickerTitleLabel.title = "From"
+                
+                if (self.activeIndexPath != nil) {
+                    self.picker?.hideShowDatePickerView(isToShow: true).configureWithTime(self.zonesArrayDate[(self.activeIndexPath?.row)!].fromDate!)
+                } else {
+                    self.picker?.hideShowDatePickerView(isToShow: true).configureWithTime(self.zonesArrayDate[self.zonesArrayDate.endIndex - 1].fromDate!)
+                }
+
```
Comment (vasakinnar) [inline Yona/Yona/Challenges/TimeFrameTimeZoneChallengeViewController.swift]: remove unwanted extra spaces

File: Yona/Yona/Challenges/TimeFrameTimeZoneChallengeViewController.swift [hunk 4]
```diff
@@ -73,160 +121,75 @@ class TimeFrameTimeZoneChallengeViewController: UIViewController {
...
      - parameter doneValue: String The number selected by the user in the picker
      - return none
      */
-    private func configureTimeZone(doneValue: String) {
-        if (activeIndexPath != nil) {
-            if doneValue.isEmpty == false {
-                let tempArr = self.generateTimeZoneArray(isFrom: isFromButton, fromToValue: zonesArray[(self.activeIndexPath?.row)!], withDoneValue: doneValue)
-                zonesArray[(self.activeIndexPath?.row)!] = tempArr
+    private func configureTimeZone(doneValue: NSDate?) {
+        print(doneValue)
```
Comment (vasakinnar) [inline Yona/Yona/Challenges/TimeFrameTimeZoneChallengeViewController.swift]: @chandanvarma009  remove log messages

File: Yona/Yona/Challenges/TimeFrameTimeZoneChallengeViewController.swift [hunk 5]
```diff
@@ -302,12 +272,116 @@ class TimeFrameTimeZoneChallengeViewController: UIViewController {
...
+            }
+        }
+    }
+    @IBAction func addTimeZoneAction(sender: AnyObject) {
+        let df = NSDateFormatter()
+        df.dateFormat = "HH:mm"
+        
+        zonesArrayString.append("\(df.stringFromDate(NSDate()))-\(df.stringFromDate(NSDate()))")
+        zonesArrayDate = zonesArrayString.converToDate()
+        print(zonesArrayDate)
```
Comment (vasakinnar) [inline Yona/Yona/Challenges/TimeFrameTimeZoneChallengeViewController.swift]: @chandanvarma009 remove log messages

File: Yona/Yona/Challenges/TimeFrameTimeZoneChallengeViewController.swift [hunk 6]
```diff
@@ -73,160 +121,75 @@ class TimeFrameTimeZoneChallengeViewController: UIViewController {
...
+            } else {
+                tempArr = self.generateTimeZoneArray(isFrom: isFromButton, fromToValue: zonesArrayDate[zonesArrayDate.endIndex - 1], withDoneValue: doneValue!)
             }
-        } else {
-            if doneValue.isEmpty == false {
-                let tempArr = self.generateTimeZoneArray(isFrom: isFromButton, fromToValue: zonesArray[zonesArray.endIndex - 1], withDoneValue: doneValue)
-                if self.isFromButton {
-                    self.zonesArray[zonesArray.endIndex - 1] = tempArr
+            
+            print(tempArr)
```
Comment (vasakinnar) [inline Yona/Yona/Challenges/TimeFrameTimeZoneChallengeViewController.swift]: @chandanvarma009 remove log messages

File: Yona/Yona/Challenges/TimeBucketChallenges.swift (hunk 1 above)
Comment (chandanvarma009) [inline Yona/Yona/Challenges/TimeBucketChallenges.swift]: Updated.

File: Yona/Yona/Challenges/TimeBucketChallenges.swift (hunk 2 above)
Comment (chandanvarma009) [inline Yona/Yona/Challenges/TimeBucketChallenges.swift]: Actually ther's no need of this condition here 😂

File: Yona/Yona/Challenges/TimeBucketChallenges.swift [hunk 7]
```diff
@@ -203,6 +198,11 @@ class TimeBucketChallenges: UIViewController,UIScrollViewDelegate {
                     Loader.Hide(self)
                 }
                 self.activityCategoriesArray = activities!
+                if self.activityCategoriesArray.count > 0 {
+                    dispatch_async(dispatch_get_main_queue()) {
+                        self.addNewGoalButton.hidden = false
+                    }
```
Comment (vasakinnar) [inline Yona/Yona/Challenges/TimeBucketChallenges.swift]: @chandanvarma009  self.addNewGoalButton.hidden = self.activityCategoriesArray.count > 0

use this instead above.

Comment (chandanvarma009) [inline Yona/Yona/Challenges/TimeBucketChallenges.swift]: updated.

**Pull request merged**